### PR TITLE
null-safety: Migrate to sound null safety

### PR DESCRIPTION
**Pull request by alikaradogan**
## What does this PR do?

- Migrates project to sound null safety.

### CODE REVIEW

File: gtubt/lib/models/user.dart [hunk 1]
```diff
@@ -6,24 +6,15 @@ part 'user.g.dart';
...
-  @JsonKey(nullable: true)
-  int year;
-
-  int id;
-  String studentId;
-
-  @JsonKey(nullable: true)
-  String phone;
-
-  @JsonKey(nullable: true)
```
Comment (gurolcay) [inline gtubt/lib/models/user.dart:25]: burasi ilginc geldi. `@JsonKey(nullable: true)` kaldirdik. Peki backend bunu donmez ise ne olacak? nasil davranis gosteriyor? veya `phone` donmedi. deserialize ederken patlamasin?

Comment (alikaradogan) [reply gtubt/lib/models/user.dart:25]: Flutter null safety ile bunlara gerek yok dedi. Artık bunlar tanımı itibariyle nullable.

**Comment by alikaradogan** [comment]
[Null safety migration](https://trello.com/c/JXxnkhXB/266-null-safety-migration)

**Review by gurolcay — CHANGES_REQUESTED** [review]
conflict

**Review by onurccn — APPROVED** [review]

**Comment by alikaradogan** [comment]
Please merge this PR before any other so we can have people writing/changing their code in respect to null-safety. 
Also I don't really want to fix more codes for null-safety. @gurolcay

**Review by gurolcay — APPROVED** [review]

**Pull request merged**